### PR TITLE
Fix SubtitleSync-TextField behavior

### PR DESCRIPTION
**Pull request by redSpoutnik**
Fix some misbehaviors I noticed upon SubtitleSync-TextField.

**Changes**

* Use ```keydown``` instead of ```keypress``` to capture 'delete' key event.
* Prevent textfield to blur while it has focus (editing subtitle offset).
* Prevent textfield to reset while it has focus.

**Issues**

None, but I can modify the textfield to make it more obvious, as pointed out in #776 (open to suggestion...)

### CODE REVIEW

File: src/components/subtitlesync/subtitlesync.js [hunk 1]
```diff
@@ -145,7 +156,9 @@ define(['playbackManager', 'layoutManager', 'text!./subtitlesync.template.html',
                         // if there is an external subtitle stream enabled
                         playbackManager.canHandleOffsetOnCurrentSubtitle(player)) {
                         // if no subtitle offset is defined
-                        if (!playbackManager.getPlayerSubtitleOffset(player)) {
+                        if (!(playbackManager.getPlayerSubtitleOffset(player) ||
+                        // or being defined (element has focus)
+                        subtitleSyncTextField.hasFocus)) {
```
Comment (dkanada) [inline src/components/subtitlesync/subtitlesync.js:161]: seems like this can be on a single line

Comment (redSpoutnik) [reply src/components/subtitlesync/subtitlesync.js:161]: I made a line break here to comment separted cases, like in:
https://github.com/jellyfin/jellyfin-web/blob/24760a1158c9575a2a96a7e65aabac12d217dbb9/src/components/subtitlesync/subtitlesync.js#L143-L146
but I can change it if you prefer.

Comment (Artiume) [reply src/components/subtitlesync/subtitlesync.js:161]: perhaps this?
 `// if showing subtitle sync is enabled && if there is an external subtitle stream enabled`

Comment (dkanada) [reply src/components/subtitlesync/subtitlesync.js:161]: Yeah, I'm not sure this makes to if-statement more readable so I'd rather have it on one line.

Comment (redSpoutnik) [reply src/components/subtitlesync/subtitlesync.js:161]: Ok, no problem.

**Pull request merged**